src/server/game/AI/CreatureAI.h

```diff
@@ -236,6 +236,12 @@ class TC_GAME_API CreatureAI : public UnitAI
         static bool IsInBounds(CreatureBoundary const& boundary, Position const* who);
         bool IsInBoundary(Position const* who = nullptr) const;
 
+        // Called when a player queries the spirit healer timer information
+        // return true to prevent automatic SMSG_AREA_SPIRIT_HEALER_TIME
+        virtual bool OnSpiritHealerQuery(Player* /*player*/) { return false; }
```

Comment:
i'd suggest to move it into a new AI class instead. We could use the SpiritGuideAI you already implemented

Reply:
done, not completely sure if that was the right way to do it but it seems to work

Reply:
to go a bit further on this. I was thinking about creating more AI classes. AreaSpiritGuideAI & PersonalSpiritGuideAI and assign them in the db with AIName. Then for the one in azure span we could use a script